plugins/intercom/stores/IntercomStore.js

```diff
@@ -0,0 +1,99 @@
+import PluginSDK from "PluginSDK";
+
+import GetSetBaseStore from "#SRC/js/stores/GetSetBaseStore";
+import {
+  DCOS_METADATA_CHANGE,
+  METADATA_CHANGE
+} from "#SRC/js/constants/EventTypes";
+import MetadataStore from "#SRC/js/stores/MetadataStore";
+
+import { INTERCOM_CHANGE } from "../constants/EventTypes";
+
+class IntercomStore extends GetSetBaseStore {
+  constructor() {
+    super(...arguments);
+
+    this.isListening = false;
```

Comment:
@nLight why we had to introduce this state? I would expect us to know from the lifecycle when there is the need to add or remove listeners without having to use conditional statements.